source/math/Quaternion.ooc

```diff
@@ -91,7 +114,73 @@ Quaternion: cover {
 		else
 			roll = asin((upYawPitch scalarProduct(upRotated) * upYawPitch z - upRotated z) / rightYawPitch z)
 
-		FloatRotation3D new(pitch, -yaw, roll)
+		FloatRotation3D new (pitch, -yaw, roll)
+	}
+	distance: func (other: Quaternion) -> Float {
+		(this - other) norm
+	}
+	rotationX: Float {
+		get {
+			result: Float
+			value := this w * this y - this z * this x
```

Comment:
Any idea what `value` is? It's used in all 3 rotation[XYZ] functions. It might be better to have a property for this, if we can figure out what it actually means.